docs/source/markdown/podman-systemd.unit.5.md

```diff
@@ -887,6 +914,114 @@ The (optional) name of the Podman volume. If this is not specified, the default
+exists on the host, pulling it if needed.
+
+Using image units allows containers and volumes to depend on images being automatically pulled. This is
+particularly interesting when using special options to control image pull.
+
+Valid options for `[Image]` are listed below:
+
+| **[Image] options**                 | **podman image pull equivalent**                |
+|-------------------------------------|-------------------------------------------------|
+| AllTags=true                        | --all-tags                                      |
+| Arch=aarch64                        | --arch=aarch64                                  |
+| AuthFile=/etc/registry/auth\.json   | --authfile=/etc/registry/auth\.json             |
+| CertDir=/etc/registery/certs        | --cert-dir=/etc/registery/certs                 |
+| Creds=myname\:mypassword            | --creds=myname\:mypassword                      |
+| DecryptionKey=/etc/registery\.key   | --decryption-key=/etc/registery\.key            |
+| Image=quay.io/centos/centos\:latest | podman image pull quay.io/centos/centos\:latest |
+| OS=windows                          | --os=windows                                    |
+| PodmanArgs=--os=linux               | --os=linux                                      |
+| Platform=linux/x86_64               | --platform=linux/x86_64                         |
+| TLSVerify=false                     | --tls-verify=false                              |
+| Variant=arm/v7                      | --variant=arm/v7                                |
+
+### `AllTags=`
+
+All tagged images in the repository are pulled.
+
+This is equivalent to the Podman `--all-tags` option
+
+### `Arch=`
+
+Override the architecture, defaults to hosts, of the image to be pulled.
+
+This is equivalent to the Podman `--arch` option
+
+### `AuthFile=`
+
+Path  of  the  authentication file.
+
+This is equivalent to the Podman `--authfile` option
+
+### `CertDir=`
+
+Use  certificates  at path (*.crt, *.cert, *.key) to connect to the registry.
+
+This is equivalent to the Podman `--cert-dir` option
+
+### `Creds=`
+
+The `[username[:password]]` to use to authenticate with the registry, if required.
+
+This is equivalent to the Podman `--creds` option
+
+### `DecryptionKey=`
+
+The `[key[:passphrase]]` to be used for decryption of images.
+
+This is equivalent to the Podman `--decryption-key` option
+
+### `Image=`
+
+The image to pull.
+It is recommended to use a fully qualified image name rather than a short name, both for
+performance and robustness reasons.
+
+The format of the name is the same as when passed to `podman pull`, so it supports e.g., using
+`:tag` or using digests guarantee a specific image version.
+
+### `OS=`
+
+Override the OS, defaults to hosts, of the image to be pulled.
+
+This is equivalent to the Podman `--os` option
+
+### `Platform=`
+
+Specify  the  platform  for  selecting the image.
+
+This is equivalent to the Podman `--platform` option
+
```

Comment:
Is there any reason to expose all of --os, --arch and --variant when a single --platform option already covers all three.
We can keep quadlet simpler here as we do not need to worry about backwards compat. I think having one way to do the same thing is better.

Reply:
I see your point. However, two comments:
1. From what I can see in the code `-variant` is not covered by `--platform`. Maybe I'm reading it wrong
2. Using `--platform` requires the user to set the OS even if they only want to set the ARCH. So, if we decide not to support something in Quadlet, I opt for not supporting `platform`. WDYT?

Reply:
That is likely incorrect docs, --platform supports a OS/ARCH/VARIANT tuple.
And the parser allows you to keep the OS empty, this works:
```
podman pull --platform /arm64/v8 alpine
```

I did not write any of this code and I don't the history of these options and maybe there is a subtle difference I am not aware of but guessing this is a mix of docker and backwards compat. I would assume @vrothberg knows more.

To me I like to keep simple and only offer one way to do the same thing. Of course that is personal preference and there is not much wrong with offering all options.

Reply:
Looking [here](https://github.com/containers/podman/blob/main/cmd/podman/images/pull.go#L144) I can see that `platform` gets parsed and the value is used for `os` and `arch`, but not `variant`. Having said that if the `arch` can be set to something like `arm64/v8` and will be treated correctly, then it should work. But, I don't know

Reply:
--platform is supposed to include --os --arch and --variant.  If it doesn't please open  an issue.

I am personally in favor of having three fields: `OS`, `Architecture` and `Variant`.  It's much easier to brain parse and more expressive.

Reply:
@vrothberg I agree. So, I've removed the support for the `Platform` key 
As for its support for variant in platform. I'm not sure if there is an issue or not and I'm not sure how to test it.